File: extra-long-factorials/extra-long-factorials.js
"use strict";

process.stdin.resume();
process.stdin.setEncoding("utf-8");

let inputString = "";
let currentLine = 0;

process.stdin.on("data", function (inputStdin) {
  inputString += inputStdin;
});

process.stdin.on("end", function () {
  inputString = inputString.split("\n");

  main();
});

function readLine() {
  return inputString[currentLine++];
}

/*
 * Complete the 'extraLongFactorials' function below.
 *
 * The function accepts INTEGER n as parameter.
 */

function extraLongFactorials(n) {
  // Write your code here
  let result = BigInt("1");
  for (let i = n; i > 0; i--) {
    result = result * BigInt(i.toString());
  }
  console.log(String(result));
}

function main() {
  const n = parseInt(readLine().trim(), 10);

  extraLongFactorials(n);
}

File: extra-long-factorials/ExtraLongFactorials.java
import java.io.*;
import java.math.*;

class Result {

    /*
     * Complete the 'extraLongFactorials' function below.
     *
     * The function accepts INTEGER n as parameter.
     */

    public static void extraLongFactorials(int n) {
    // Write your code here
        BigInteger result = BigInteger.ONE;
        for (int i = n; i > 0; i--) {
            result = result.multiply(BigInteger.valueOf(i));
        }
        System.out.print(result);
    }

}

public class ExtraLongFactorials {
    public static void main(String[] args) throws IOException {
        BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(System.in));

        int n = Integer.parseInt(bufferedReader.readLine().trim());

        Result.extraLongFactorials(n);

        bufferedReader.close();
    }
}
